parameter/DefaultElementLibrary.h

```diff
@@ -46,7 +47,13 @@ class CDefaultElementLibrary: public CElementLibrary
 {
 public:
 
-    explicit CDefaultElementLibrary(bool bEnableDefaultMechanism = true);
+    /**
+     * Class constructor
+     *
+     * @param[in] logger the logger provided by the client
+     * @param[in] bEnableDefaultMechanism boolean allowing to activate VirtualSubsystem use
+     */
+    explicit CDefaultElementLibrary(core::log::ILogger& logger, bool bEnableDefaultMechanism = true);
```

Comment:
Shorter (same) example:
~~~ {cpp}
struct I1 {          I1(int a=0       ) {} };
struct I2 {          I2(int a, int b=0) {} };
struct E1 { explicit E1(int a=0       ) {} };
struct E2 { explicit E2(int a, int b=0) {} };

template <class T> void f(T) {};

f<I1>();  // Error: too few arguments to function call, expected 1, have 0
f<I1>(1); // OK
f<I2>(1); // OK

f<E1>();  // Error: too few arguments to function call, expected 1, have 0
f<E1>(1); // Error: no viable conversion from 'int' to 'E'
f<E2>(1); // Error: no viable conversion from 'int' to 'E'
~~~